src/io/mod.rs

```diff
@@ -10,16 +10,16 @@ pub struct Obfuscation<M: PolyMatrix> {
     pub enc_hardcoded_key: M,
     pub encodings_init: Vec<BggEncoding<M>>,
     pub p_init: M,
-    pub m_preimages: Vec<(M, M)>,
-    pub n_preimages: Vec<(M, M)>,
-    pub k_preimages: Vec<(M, M)>,
+    pub m_preimages: Vec<Vec<M>>,
+    pub n_preimages: Vec<Vec<M>>,
+    pub k_preimages: Vec<Vec<M>>,
     pub final_preimage: M,
     #[cfg(feature = "test")]
     pub s_init: M,
     #[cfg(feature = "test")]
     pub t_bar: <M as PolyMatrix>::P,
     #[cfg(feature = "test")]
-    pub bs: Vec<(M, M, M)>,
+    pub bs: Vec<Vec<M>>,
```

Comment:
@enricobottazzi Do you have any reason of this change?

Reply:
Main reason is that in the refactored code, we don't compute all the `bs` at once but actually compute them for each `index` and for each `{0, 1, star}` in different moments so it's easier to place them into a preallocated vector of vector of matrices rather than constructing a tuple.